src/components/code/code_block.tsx

```diff
@@ -232,28 +232,35 @@ export const EuiCodeBlock: FunctionComponent<EuiCodeBlockProps> = ({
     [preClasses, onKeyDown]
   );
 
-  const optionalStyles: CSSProperties = {};
-
-  if (overflowHeight) {
-    const property =
-      typeof overflowHeight === 'string' ? 'height' : 'maxHeight';
-    optionalStyles[property] = overflowHeight;
-  }
+  const optionalStyles: CSSProperties = useMemo(() => {
+    if (overflowHeight) {
+      const property =
+        typeof overflowHeight === 'string' ? 'height' : 'maxHeight';
+      return {
+        [property]: overflowHeight,
+      };
+    }
+    return {};
```

Comment:
super super optional: it might be nice to return undefined here so style isn't applied as an attribute at all. no worries if you'd rather not though 
```suggestion
    return undefined;
```